src/test/java/org/broadinstitute/hellbender/utils/nio/LineIteratorTest.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class LineIteratorTest {
+
+  @Test
+  public void testLineIteratorInForEach() throws IOException {
+    try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+      final Path path = jimfs.getPath("test.txt");
+      try (BufferedWriter bufferedWriter = Files.newBufferedWriter(path)) {
+        bufferedWriter.write("Hello world\n");
+        bufferedWriter.write("What's new?");
+      }
+      ArrayList<String> got = new ArrayList<>();
+      for (String s: new LineIterator(path)) {
+        got.add(s);
+      }
+      Assert.assertEquals(got.size(), 2);
+      Assert.assertEquals(got.get(0), "Hello world");
+      Assert.assertEquals(got.get(1), "What's new?");
+    }
+  }
+
+  @Test
+  public void testLineIteratorAsResource() throws IOException {
+    try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+      final Path path = jimfs.getPath("test.txt");
+      try (BufferedWriter bufferedWriter = Files.newBufferedWriter(path)) {
+        bufferedWriter.write("Hello world\n");
+        bufferedWriter.write("What's new?");
+      }
+      ArrayList<String> got = new ArrayList<>();
+      try (LineIterator lines = new LineIterator(path)) {
+        lines.forEach(s -> got.add(s));
+      }
+      Assert.assertEquals(got.size(), 2);
+      Assert.assertEquals(got.get(0), "Hello world");
+      Assert.assertEquals(got.get(1), "What's new?");
+    }
+  }
+
```

Comment:
Can you add a test case that reads a text file from our test GCS bucket (marked as `groups={"bucket"}`)?

Reply:
I'm not sure this is the best use of our resources. `PathLineIterator` is really a thin wrapper around `Files.lines`, so this would be essentially testing Java's standard library.

I added the test and put it inside of GcsNioIntegrationTest (since it touches outside resources and checks the cloud case).